### PR TITLE
Store creation: Plan purchase

### PR DESCRIPTION
**Pull request by 0nko**
This PR implements the plan purchase step of the store creation.

**To test:**
1. Start the store creation flow
2. When you get to the domain selection and tap on Continue, verify the plan purchase screen is shown
3. Tapping on the X button goes back to the previous screen

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    private val _viewState = savedState.getStateFlow<ViewState>(this, LoadingState)
+    val viewState = _viewState.asLiveData()
+
+    init {
+        loadPlan()
+    }
+
+    private fun loadPlan() {
+        launch {
+            delay(1000)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt]: hmm, why do you need this delay? Testing purposes for loading state? Can be removed?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansScreen.kt [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    viewState: PlanState
+) {
+    val systemUiController = rememberSystemUiController()
+    val color = colorResource(id = color.woo_purple_90)
+    SideEffect {
+        systemUiController.setSystemBarsColor(
+            color = color,
+            darkIcons = false
+        )
+    }
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansScreen.kt]: hmm there's a problem with this code. When navigating back from this screen the status bar and navigation bar will still be in `woo_purple_90` for all the fragments of the same activity.

I spent some time looking into this. Not the biggest fan of the code I'm about to suggest as is kind of hacky but is the cleanest approach I've found to make it work. We could use `DisposableEffect` tied to the view Model scope, something like this:
```kotlin
    val systemUiController = rememberSystemUiController()
    val wooDarkPurple = colorResource(id = color.woo_purple_90)
    val statusBarColor = colorResource(id = color.color_status_bar)
    DisposableEffect(key1 = viewModel) {
        systemUiController.setSystemBarsColor(
            color = wooDarkPurple,
        )
        onDispose {
            systemUiController.setSystemBarsColor(
                color = statusBarColor,
            )
        }
    }
```

Also you can get rid of ` darkIcons = false` and leave it to use `darkIcons: Boolean = color.luminance() > 0.5f` which will set them light for this screen.  



https://user-images.githubusercontent.com/2663464/201730567-4aa3e18c-3076-4e51-95bc-ff51f2969559.mp4

There's still some color flickering in the status bar when navigating back and forth but I guess we can live with that. Let me know what you think

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt]: Oh yes, I used this to test the loading when when I tried fetching the plan data. I'll get rid of this.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansScreen.kt (hunk 2 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansScreen.kt]: This works great, thanks for the suggestion!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt (hunk 1 above)
Comment (0nko) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/plans/PlansViewModel.kt]: I realized I already fixed this but forgot to push it 🤦

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Look good and works well! Thanks for applying the changes @0nko  👍🏼 
:shipit:

**Pull request merged**